README.md

```diff
@@ -82,132 +94,451 @@ integration_types:
             default_value: "1024"
 ```
 
+
 * `id` - The symbolic identifier used to select the integration type within the DIT.
 * `name` - A user friendly name for the integration.
 * `description` - A description of what the integration does.
 * `entrypoint` - The package qualified name of the entrypoint function.
-* `env_class` - The package qualifies name of the environment class.
-* `runtime` - Always `python-direct`.
-* `fields` - A list of configuration fields. These are passed into the integration at runtime via correspondingly named fields of an instance of the `env_class`.
+* `env_class` - The package qualified name of the class used to contain the integration's _environment_.
+* `fields` - A list of [configuration fields](https://github.com/digital-asset/daml-dit-if#integration-configuration-arguments) that users will be able to enter through the console when configuring an integration. These will be passed into the integration instance at runtime via an instance of `env_class`.
+
+The `entrypoint` and `env_class` fields identify by name the two
+Python structures that represent the runtime definition of an
+integration type's implemenation.
+
+The first, `entrypoint`, is required for all integration types and
+names a function the framework calls when starting a new instance of
+an integration. During the entrypoint function, integrations are able
+to register handlers for various sort of events (ledger, web, and
+otherwise), start coroutines, and access integration configuration
+arguments specified through the Daml Hub console UI. The syntax for
+`entrypoint` is `$PYTHON_PACKAGE_NAME:$FUNCTION_NAME`, and the function
+itself must have the following signature.
 
-The Python definition of the entrypoint is this:
+```python
+from daml_dit_if.api import IntegrationEvents
 
+def integration_ledger_event_log_main(
+        env: 'IntegrationLedgerEventLogEnv',
+        events: 'IntegrationEvents'):
+
+```
+
+The `events` argument is an instance of [`IntegrationEvents`](https://github.com/digital-asset/daml-dit-if/blob/master/daml_dit_if/api/__init__.py#L272)
+containing a number of [function decorators](https://www.python.org/dev/peps/pep-0318/)
+that can be used to register event handlers. This represents the bulk
```

Comment:
It's only a few paragraphs down, but possibly link "event handlers" to the `## Integration Event Handlers` section?

Reply:
Great idea.